modules/consensus/database_helpers_test.go

```diff
@@ -0,0 +1,476 @@
+	scoBytes, err := db.getItem(SiacoinOutputs, id)
+	if err != nil {
+		panic(err)
+	}
+	var sco types.SiacoinOutput
+	err = encoding.Unmarshal(scoBytes, &sco)
+	if build.DEBUG && err != nil {
+		panic(err)
+	}
+	return sco
+}
+
+// forEachFileContracts applies a function to each (file contract id, filecontract)
+// pair in the consensus set
+func (db *setDB) forEachFileContracts(fn func(k types.FileContractID, v types.FileContract)) {
+	db.forEachItem(FileContracts, func(kb, vb []byte) error {
+		var key types.FileContractID
+		var value types.FileContract
+		err := encoding.Unmarshal(kb, &key)
+		if err != nil {
+			return err
+		}
+		err = encoding.Unmarshal(vb, &value)
+		if err != nil {
+			return err
+		}
+		fn(key, value)
+		return nil
+	})
+}
+
+// rmSiafundOutputs removes a siafund output from the database
+func (db *setDB) rmSiafundOutputs(id types.SiafundOutputID) error {
+	return db.rmItem(SiafundOutputs, id)
+}
+
+// inSiafundOutputs is a wrapper around inBucket which returns a true
+// if an output with the given id is in the database
+func (db *setDB) inSiafundOutputs(id types.SiafundOutputID) bool {
+	return db.inBucket(SiafundOutputs, id)
+}
+
+// getSiafundOutputs is a wrapper around getItem which decodes the
+// result into a siafundOutput
+func (db *setDB) getSiafundOutputs(id types.SiafundOutputID) types.SiafundOutput {
+	sfoBytes, err := db.getItem(SiafundOutputs, id)
+	if build.DEBUG && err != nil {
+		panic(err)
+	}
+	var sfo types.SiafundOutput
+	err = encoding.Unmarshal(sfoBytes, &sfo)
+	if build.DEBUG && err != nil {
+		panic(err)
+	}
+	return sfo
+}
+
+// rmItem removes an item from a bucket
+func (db *setDB) rmItem(bucket []byte, key interface{}) error {
+	k := encoding.Marshal(key)
+	return db.Update(func(tx *bolt.Tx) error {
+		b := tx.Bucket(bucket)
+		if build.DEBUG {
+			// Sanity check to make sure the bucket exists.
+			if b == nil {
+				panic(errNilBucket)
+			}
+			// Sanity check to make sure you are deleting an item that exists
+			item := b.Get(k)
+			if item == nil {
+				panic(errNilItem)
+			}
+		}
+		return b.Delete(k)
+	})
+}
+
+// inBucket checks if an item with the given key is in the bucket
+func (db *setDB) inBucket(bucket []byte, key interface{}) bool {
+	exists, err := db.Exists(bucket, encoding.Marshal(key))
+	if build.DEBUG && err != nil {
+		panic(err)
+	}
+	return exists
+}
+
+// lenBucket is a simple wrapper for bucketSize that panics on error
+func (db *setDB) lenBucket(bucket []byte) uint64 {
+	s, err := db.BucketSize(bucket)
+	if build.DEBUG && err != nil {
+		panic(err)
+	}
+	return s
+}
+
+// forEachItem runs a given function on every element in a given
+// bucket name, and will panic on any error
+func (db *setDB) forEachItem(bucket []byte, fn func(k, v []byte) error) {
+	err := db.View(func(tx *bolt.Tx) error {
+		b := tx.Bucket(bucket)
+		if build.DEBUG && b == nil {
+			panic(errNilBucket)
+		}
+		return b.ForEach(fn)
+	})
+	if err != nil {
+		panic(err)
+	}
```

Comment:
unqualified panic

there are a whole lot of functions here. Should grep around and make sure they are all being used. Some could probably be combined.